beebop/app.py

```diff
@@ -247,8 +253,21 @@ def generate_microreact_url_internal(microreact_api_new_url,
     r = requests.post(microreact_api_new_url,
                       data=json.dumps(json_microreact),
                       headers=headers)
-    url = r.json()['url']
-    return jsonify(response_success({"cluster": cluster, "url": url}))
+    if r.status_code == 500:
+        return jsonify(error=response_failure({
+            "error": "Wrong Token",
+            "detail": """
+            Microreact reported Internal Server Error.
+            Most likely Token is invalid!"""
+            })), 500
+    elif r.status_code == 404:
+        return jsonify(error=response_failure({
+            "error": "Resource not found",
+            "detail": "Cannot reach Microreact API"
+            })), 404
+    elif r.status_code == 200:
+        url = r.json()['url']
+        return jsonify(response_success({"cluster": cluster, "url": url}))
```

Comment:
What happens when we get an other response type? Often people will put in a check for anything >= 400, but there should be an else clause here or if microreact returns something unexpected teh app will behave weirdly.

Consider

```
else:
  return jsonify(error=response_failure({"error": "Unknown error", "detail": (save a bunch of stuff from r here ideally)})), 500
```

perhaps? When saving out the error you might make sure that the http code at least makes it through, plus perhaps the content if it's text? It's not hugely likely but you might get a 400 if you give MR ill-formatted data (your current 500 test might end up reported as a 400 by MR in future for example)